src/Compiler/FSComp.txt

```diff
@@ -1708,4 +1708,5 @@ featureAccessorFunctionShorthand,"underscore dot shorthand for accessor only fun
 3569,chkNotTailRecursive,"The member or function '%s' has the 'TailCallAttribute' attribute, but is not being used in a tail recursive way."
 3570,tcStaticBindingInExtrinsicAugmentation,"Static bindings cannot be added to extrinsic augmentations. Consider using a 'static member' instead."
 3571,pickleFsharpCoreBackwardsCompatible,"Newly added pickle state cannot be used in FSharp.Core, since it must be working in older compilers+tooling as well. The time window is at least 3 years after feature introduction. Violation: %s . Context: \n %s "
-3577,tcOverrideUsesMultipleArgumentsInsteadOfTuple,"This override takes a tuple instead of multiple arguments. Try to add an additional layer of parentheses at the method definition (e.g. 'member _.Foo((x, y))'), or remove parentheses at the abstract method declaration (e.g. 'abstract member Foo: 'a * 'b -> 'c')."
\ No newline at end of file
+3577,tcOverrideUsesMultipleArgumentsInsteadOfTuple,"This override takes a tuple instead of multiple arguments. Try to add an additional layer of parentheses at the method definition (e.g. 'member _.Foo((x, y))'), or remove parentheses at the abstract method declaration (e.g. 'abstract member Foo: 'a * 'b -> 'c')."
+3578,chkCopyUpdateSyntaxInAnonRecords,"Label '%s' is part of anonymous record. Use {{| %s with %s = ... |}} instead."
```

Comment:
```suggestion
3578,chkCopyUpdateSyntaxInAnonRecords,"Value '%s' is an anonymous record. Use syntax {{| %s with %s = ... |}} to update it."
```

+ switch the arguments.
Imo the name of the anon record value should  be the subject in this sentence.

Reply:
@T-Gro, probably "use the following syntax ... to update it"? My english is not very good, maybe @KathleenDollard can help us here?

Reply:
Happy to adjust the error message. I was hoping for a better error suggested :).

Reply:
Do we actually use term `Label` for record fields anywhere in the docs?

Reply:
@auduchinok I will wait to see what is the final suggested message to update it :)

Reply:
This is still a bit confusing for me, because the problem isn't with `'ff'` but rather with `expr`. I think Tomas' proposal makes more sense. Is it possible to get the name of `expr` here? 

Also if the user is already using correct `with` syntax, we could be even more focused on the actual problem, e.g. `'expr' is an anonymous record, use {|...|} instead of {...}`.

Reply:
> 

In my initial implementation I had the `Ident` name from `expr` but as Eugene pointed out [here](https://github.com/dotnet/fsharp/pull/15598#discussion_r1272201892). I was only checking for one specific shape of the `SynExpr` and that will eliminate the error reporting for the rest of the cases where an anonymous record value can be obtained like from a property (x.Prop), a function call (f ()), an indexer (x.[0], or x[0]), etc. 

I think are more general approach needs to be used here so that is why I'm using `expr`

Reply:
I suppose we can say `This expression is an anonymous record, use {|...|} instead of {...}.` if the whole `{ ... }` is underlined. I would prefer that over mentioning a particular field from the record. What do others think?

Reply:
This PR uses a more accurate error range based on the annon record field that is being used.

Reply:
That doesn't seem very accurate to me, seeing that the fields have nothing to do with the error.

Reply:
I mean they are related as it is trying to update an annon record using the record syntax. and we now that because 2 of the fields belong to an annon record. ?